packages/utils/telemetry-utils/src/mockLogger.ts

```diff
@@ -141,7 +141,22 @@ ${JSON.stringify(actualEvents)}`);
 		actual: ITelemetryBaseEvent,
 		expected: Omit<ITelemetryBaseEvent, "category">,
 	): boolean {
-		const masked = { ...actual, ...expected };
-		return JSON.stringify(masked) === JSON.stringify(actual);
+		// "details" is used in a lot of telemetry logs to group a bunch of properties together and stringify them.
+		// Some of the properties in the expected event may be inside "details". So, if the expected event does not have
+		// "details", extract the properties from "details" in the actual event and inline them in the actual event.
+		const { details, ...actualForMatching } = actual;
+		let detailsExpanded = { details };
```

Comment:
Doesn't this just give you an object like `{ details }`...?  And later when you do `...detailsExpanded` that would just give `details`.  In other words `actualExpanded` is identical to `actual`, I think?  Or what am I missing?

Reply:
Oh I understand now.  This is modified below in the `if` block.